crates/red_knot_test/src/lib.rs

```diff
@@ -268,3 +306,36 @@ struct TestFile {
     // Offset of the backticks that starts the code block in the Markdown file
     backtick_offset: TextSize,
 }
+
+fn create_diagnostic_snapshot<D: Diagnostic>(
+    db: &mut db::Db,
+    test: &parser::MarkdownTest,
+    diagnostics: impl IntoIterator<Item = D>,
+) -> String {
+    // TODO(ag): Do something better than requiring this
+    // global state to be twiddled everywhere.
+    colored::control::set_override(false);
```

Comment:
(see also: https://github.com/astral-sh/ruff/issues/13939)

Reply:
Blech, yeah. I think we need more fine grained color configuration than just a global switch. The global switch makes it very easy to abuse.